### ISSUE TITLE
Is it possible to resize images  only when they exceed max-Size

### ISSUE THREAD

**Issue by nevyen** [danialfarid/ng-file-upload#1184]
Hey,
maybe this question is too simple, but I don't get it. 

Is it possible to resize images only when the exceed max-Size or max-height. 

And how to resize them to a given width by keep the ratio ? 

thanks 
Dominic

**Comment by danialfarid** [comment]
It would always keep the ratio, so you can just set the width and the height will be calculated depending on centerCrop value.
For now the resize will be applied to all images. You can use `Upload.imageDimension` and `Upload.resize()` service to apply your filtering before resizing.

**Comment by nevyen** [comment]
Thanks for this hint I will try it. And many add a Wiki page

**Comment by nasir-mps** [comment]
I also have similar scenario, I want to scale down an image, if its larger than specific height or width. Once, we have resize it, we will upload it to server.  How Upload.resize() works?

var resizedImage = Upload.resize(imagefile,100,100, 0.8,'image/jpg','1:2',true);

Upload.http({
        url: test.com/url,
        data: resizedImage ,
                        method: "PUT",
                        headers : {
                            'content-type': file.type
                        },
                        transformRequest: function(data, headersGetter) {
                            var headers = headersGetter();
                            delete headers.Authorization;
                            return data;
                        }
                    });

**Comment by danialfarid** [comment]
I updated the readme and added docs for `Upload.resize()`

**Comment by gmisiolek** [comment]
@danialfarid If I am not wrong, fixing this issue has only few lines of code :D

In resize.js I changed lines 41 - 46:
```javascript
if (!width) {
  width = imageElement.width;
} else {
  if (width > imageElement.width) {
    width = imageElement.width;
  }
}
if (!height) {
  height = imageElement.height;
} else {
  if (height > imageElement.height) {
    height = imageElement.height;
  }
}
```
It works for me. What do you think?

**Comment by danialfarid** [comment]
Fixed at 11.0.0.
You can use `ngf-resize-if`
